portality/static/widget/fqw_js.js

```diff
@@ -0,0 +1,7404 @@
+/** @namespace */
```

Comment:
I couldn't see anywhere where this file was autogenerated, so I'm guessing this is manually created?  Since it's basically copies of all the core dependencies, we should have this compiled from those dependencies.  There's a script that does this already for the old version which you can modify, it's just a few lines of bash that cats all the files together, see `fixed_query_build.sh`

Reply:
It is not exactly the copy of all the core dependencies as it is much smaller - I have removed all the methods from the core that are not used by the widget. For doaj.js for example it is quite a lot.

Reply:
There's a size vs maintenance trade-off here, and I would come down on the side of ease of maintenance.  Otherwise any change that we make to `doaj.js` we need to go and check if that change is relevant to the fqw and make the change there.  Otherwise, we just introduce a build step which recompiles fqw on release and we no longer have to worry about it.

Reply:
You are right, this is quite a burden in terms of maintenance. But we need to remember that this is the js that clients machine needs to upload. I do understand your point of view but I am not fully convinced that simply compiling everything is the way to go. Cause why not just import all for js files?

Let me think about it.

Reply:
What was your final position on this in the end?  Does the updated build script above mean that this is now being compiled from the other files?

Reply:
Autogenerated as it used to be. Even though I don't like it ;)